packages/container/libraries/strongly-typed/src/container.ts

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unnecessary-type-parameters */
+import { Container, type interfaces } from 'inversify';
+
+type IfAny<T, TYes, TNo> = 0 extends 1 & T ? TYes : TNo;
+
+type BindingMapProperty = string | symbol;
+type BindingMap = Record<BindingMapProperty, any>;
+type BindingMapKey<T extends BindingMap> = keyof T & BindingMapProperty;
+type MappedServiceIdentifier<T extends BindingMap> = IfAny<
+  T,
+  interfaces.ServiceIdentifier,
+  BindingMapKey<T>
```

Comment:
```suggestion
  keyof T
```
I managed to compile the source code with this change :)